### PR TITLE
add docker support

### PR DESCRIPTION
**Pull request by hobbypunk90**
# Description

i add basically support for docker. 
with `docker build -t test/bt-mqtt-gateway .` you can build the container
and for example with `docker run -d --name bt-mqtt --network=host -v $PWD:/config test/bt-mqtt-gateway` you can run it as detached docker container 🙂

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    cat /tmp/requirements.txt && \
+    pip install -r /tmp/requirements.txt && \
+    rm /tmp/requirements.txt && \
+    apk del --no-cache bluez-dev musl-dev gcc make git glib-dev linux-headers grep python2
+
+ADD ./start.sh /start.sh
+RUN chmod +x /start.sh
+
+ENV DEBUG false
+ENV TZ Europe/Berlin
```
Comment (zewelor) [inline Dockerfile]: Is this needed ?

File: start.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+ #!/bin/sh
+
+ if ! [ -f '/config/config.yaml' ]; then
+    echo "There is no config.yaml! An example is created."
+    cp /application/config.yaml.example /config/config.yaml.example
```
Comment (zewelor) [inline start.sh]: ```cp /application/config.yaml.example /config/config.yaml```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+VOLUME ["/config"]
+
+ENTRYPOINT ["/bin/sh", "-c", "/start.sh"]
```
Comment (zewelor) [inline Dockerfile]: Maybe entrypoint should be script, and use docker restart policy to keep it alive ?

Comment (hobbypunk90) [reply Dockerfile]: the start script exits with the status code of the gateway 😉 with the commit 2f5d6f0 also on debug mode 😅

File: Dockerfile (hunk 1 above)
Comment (hobbypunk90) [reply Dockerfile]: the timezone? im not sure, i'll try without tomorrow

File: start.sh (hunk 2 above)
Comment (hobbypunk90) [reply start.sh]: i'm not sure 🤔
if we copy the example direkt to config.yaml, the next time the container is started it tries to run with the example config.
Nobody want this, thats why i copied it to *.example.

special, if you start the container with restart=unless-stopped (which is my default 😁) copying it direct to config.yaml isn't that nice 😉😁

File: Dockerfile (hunk 1 above)
Comment (zewelor) [reply Dockerfile]: Yes, I assume not everyone will have the same timezone :D ,can't docker use timezone from host ?

File: start.sh (hunk 2 above)
Comment (zewelor) [reply start.sh]: Oh I read it wrong, i thought it was supposed to create default config, when it doesn't exist

File: Dockerfile (hunk 1 above)
Comment (hobbypunk90) [reply Dockerfile]: it can, if every system would save the TZ at the same place 🙈😂
but it seam that we can drop it

File: start.sh (hunk 2 above)
Comment (hobbypunk90) [reply start.sh]: oh, yes and no, it creates a sample config at build time.
i think, depending on the overlay driver the user never see this file 🤔

**Comment by zewelor** [comment]
Added docker hub autobuilds and built first image: https://hub.docker.com/r/zewelor/bt-mqtt-gateway

**Pull request merged**